changelog_unreleased/typescript/16327.md

```diff
@@ -0,0 +1,22 @@
+#### Improve formatting of arrow functions with TSAsExpression body (#16327 by @neelance)
+
+<!-- prettier-ignore -->
+```ts
+// Input
+call(() => ({
+  foo: 1,
+}) as const);
+
+// Prettier stable
+call(
+  () =>
+    ({
+      foo: 1,
+    }) as const,
+);
+
+// Prettier main
+call(() => ({
+  foo: 1,
+}) as const);
+```
```

Comment:
I feel the stable output is more readable, but I'm fine with this change if others think it's an improvement.

Reply:
I can understand why you feel that way with this simple example. If it helps I could present a real-world example from our codebase which shows a much more profound impact.